### ISSUE TITLE
[BUG] Controllers not supported with Android 12

### ISSUE THREAD

**Issue by Timbo303** [PojavLauncherTeam/PojavLauncher#2254]
### Describe the bug

Any Controller I try crashes the entire game after one single controller input. Everything else works just fine otherwise.

java.lang.NullPointerException: Attempt to invoke virtual method 'void java.lang.reflect.Field.setInt(java.lang.Object, int)' on a null object reference
	at net.kdt.pojavlaunch.customcontrols.gamepad.GamepadDpad.setDummyEventKeycode(GamepadDpad.java:76)
	at net.kdt.pojavlaunch.customcontrols.gamepad.GamepadDpad.convertEvent(GamepadDpad.java:67)
	at net.kdt.pojavlaunch.customcontrols.gamepad.Gamepad.update(Gamepad.java:158)
	at net.kdt.pojavlaunch.BaseMainActivity.dispatchGenericMotionEvent(BaseMainActivity.java:593)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchGenericMotionEvent(WindowCallbackWrapper.java:79)
	at com.android.internal.policy.DecorView.dispatchGenericMotionEvent(DecorView.java:958)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processGenericMotionEvent(ViewRootImpl.java:7724)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:7442)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6773)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6830)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6796)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6994)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6804)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:7051)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6777)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6830)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6796)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6804)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6777)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6830)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6796)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:7027)
	at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:7214)
	at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:3387)
	at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:2948)
	at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:2939)
	at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:3364)
	at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:154)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:335)
	at android.os.Looper.loopOnce(Looper.java:186)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8582)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)

### Steps To Reproduce

```markdown
1. Start game with android 12 and any controller connected to phone
2. Press any input after startup
3. Grats your game crashes.
```

### Expected Behavior

Basically not a crash

### Platform

```markdown
- Device model: Samsung S21 5G
- CPU architecture: Snapdragon 888
- Android version: 12
- PojavLauncher version: Any
```

**Comment by Mathias-Boulay** [comment]
duplicate from #2217, but at least it confirms the common denominator is android 12 behavior.
Sadly I don't have an android 12 device, so looking at this matter is going to take some time

**Comment by Timbo303** [comment]
Get a samsung s21 or something it recently got the update.

**Comment by Mathias-Boulay** [comment]
@Timbo303 no problem, just buy me one 😂

**Comment by ajit850** [comment]
> @Timbo303 no problem, just buy me one 😂

😂😂😂😂

**Comment by Muvkina** [comment]
Also happening here android 12 s21 ultra

**Comment by Muvkina** [comment]
New control revision doesn't seem to fix it by the way

**Comment by Mathias-Boulay** [comment]
Known, the small cleanup PR will though
#2267

**Comment by Muvkina** [comment]
Ok awesome! Is there any github action to get a build of that pr?

**Comment by Mathias-Boulay** [comment]
Not yet, but you won't have to wait too much

**Comment by Mathias-Boulay** [comment]
merged
